engine/class_modules/apl/warlock.cpp

```diff
@@ -177,7 +177,9 @@ void demonology( player_t* p )
   default_->add_action( "demonic_strength" );
   default_->add_action( "bilescourge_bombers,if=!pet.demonic_tyrant.active" );
   default_->add_action( "shadow_bolt,if=soul_shard<5&talent.fel_covenant&buff.fel_covenant.remains<5" );
-  default_->add_action( "implosion,if=active_enemies>2&buff.wild_imps.stack>=6" );
+  default_->add_action( "implosion,if=two_cast_imps>0&buff.tyrant.down&active_enemies>1+(talent.sacrificed_souls.enabled)" );
+  default_->add_action( "implosion,if=buff.wild_imps.stack>9&buff.tyrant.up&active_enemies>2+(1*talent.sacrificed_souls.enabled)&cooldown.call_dreadstalkers.remains>17&talent.the_expendables" );
+  default_->add_action( "implosion,if=active_enemies=1&last_cast_imps>0&buff.tyrant.down&talent.imp_gang_boss.enabled&!talent.sacrificed_souls" );
```

Comment:
Suggestion: see if "igb_ratio" expression can assist you here (recently added)
doc: https://github.com/simulationcraft/simc/wiki/Warlocks

Reply:
To provide an update, for now, i did not manage to find a igb_ratio or combination of igb_ratio + other conditionals that created better results in single target.